saas/management/commands/report_weekly_revenue.py

```diff
@@ -49,172 +49,236 @@ def add_arguments(self, parser):
-
-        _, payment_amounts, payments_unit = \
-            aggregate_transactions_by_period(
-                provider, Transaction.RECEIVABLE,
-                orig='dest', dest='dest',
-                orig_account=Transaction.BACKLOG,
-                orig_organization=provider,
-                date_periods=prev_week)
-        _, payment_amounts_prev_year, _ = \
-            aggregate_transactions_by_period(
-                provider, Transaction.RECEIVABLE,
-                orig='dest', dest='dest',
-                orig_account=Transaction.BACKLOG,
-                orig_organization=provider,
-                date_periods=prev_year)
-
-        _, refund_amounts, refund_unit = \
-            aggregate_transactions_by_period(
-                provider, Transaction.REFUND,
-                orig='dest', dest='dest',
-                date_periods=prev_week)
-        _, refund_amounts_prev_year, _ = \
-            aggregate_transactions_by_period(
-                provider, Transaction.REFUND,
-                orig='dest', dest='dest',
-                date_periods=prev_year)
+    def get_perf_data(provider, prev_periods, prev_year_periods, period_type):
+        # pylint:disable=too-many-locals
+        account_table, _, _, table_unit = aggregate_transactions_change_by_period(
+            provider, Transaction.RECEIVABLE, account_title='Sales',
+            orig='orig', dest='dest', date_periods=prev_periods
+        )
+        account_table_prev_year, _, _, _ = aggregate_transactions_change_by_period(
+            provider, Transaction.RECEIVABLE, account_title='Sales',
+            orig='orig', dest='dest', date_periods=prev_year_periods
+        )
 
-        unit = settings.DEFAULT_UNIT
+        _, payment_amounts, payments_unit = aggregate_transactions_by_period(
+            provider, Transaction.RECEIVABLE,
+            orig='dest', dest='dest', orig_account=Transaction.BACKLOG,
+            orig_organization=provider, date_periods=prev_periods
+        )
+        _, payment_amounts_prev_year, _ = aggregate_transactions_by_period(
+            provider, Transaction.RECEIVABLE,
+            orig='dest', dest='dest', orig_account=Transaction.BACKLOG,
+            orig_organization=provider, date_periods=prev_year_periods
+        )
+
+        _, refund_amounts, refund_unit = aggregate_transactions_by_period(
+            provider, Transaction.REFUND,
+            orig='dest', dest='dest', date_periods=prev_periods
+        )
+        _, refund_amounts_prev_year, _ = aggregate_transactions_by_period(
+            provider, Transaction.REFUND,
+            orig='dest', dest='dest', date_periods=prev_year_periods
+        )
 
+        unit = settings.DEFAULT_UNIT
         units = get_different_units(table_unit, payments_unit, refund_unit)
 
         if len(units) > 1:
-            LOGGER.error("different units in get_weekly_perf_data: %s", units)
+            LOGGER.error("different units in get_%s_perf_data: %s", period_type, units)
```

Comment:
more like:
 `LOGGER.error("different units in get_perf_data(period_type=%s): %s", period_type, units)`